xdsl/pattern_rewriter.py

```diff
@@ -726,7 +731,11 @@ def _handle_operation_removal(self, op: Operation) -> None:
         """Handle removal of an operation."""
         if self.apply_recursively:
             self._add_operands_to_worklist(op.operands)
-        self._worklist.remove(op)
+        if op.regions:
+            for sub_op in op.walk():
+                self._worklist.remove(sub_op)
+        else:
+            self._worklist.remove(op)
```

Comment:
```suggestion
        self._worklist.remove(op)
```
Surely the operation itself still needs to be removed when it has regions?